src/Core/src/Handlers/Entry/EntryHandler.Android.cs

```diff
@@ -84,6 +107,32 @@ public static void MapReturnType(EntryHandler handler, IEntry entry)
 			handler.TypedNativeView?.UpdateReturnType(entry);
 		}
 
+		public static void MapClearButtonVisibility(EntryHandler handler, IEntry entry)
+		{
+			handler.TypedNativeView?.UpdateClearButtonVisibility(entry, ClearButtonDrawable);
+		}
+
+		void OnFocusedChange(bool hasFocus)
+		{
+			if (TypedNativeView == null || VirtualView == null)
+				return;
+
+			// This will eliminate additional native property setting if not required.
+			if (VirtualView.ClearButtonVisibility == ClearButtonVisibility.WhileEditing)
+				TypedNativeView?.UpdateClearButtonVisibility(VirtualView, ClearButtonDrawable);
+		}
+
+		bool OnTouch(MotionEvent? motionEvent)
+		{
+			if (TypedNativeView == null || VirtualView == null)
+				return false;
+
+			// Update clear button visibility.
+			TypedNativeView.UpdateClearButtonVisibility(VirtualView, ClearButtonDrawable);
```

Comment:
Do we need this here? Since the clear button will get handled when the text changes, maybe all that needs to be done is just test `Text = ""` and then let that event set the visibility? Not sure if that actually works, but a thought.

Reply:
Good catch. I was going to skip IOnFocusChangeListener at first and wait for general implementation in IView layer. It was there for testing and kind of mimic the behavior of Focus, then I implemented IOnFocusChangeListener and forgot to remove it.